modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/TaskActivityService.java

```diff
@@ -109,20 +109,22 @@
      *
      * @param activityTypeSet the type of activities
      * @param queryParams query parameters to use while retrieving
+     * @param grouping
      * @return the list of all activities
      */
-    List<TaskActivity> getAllActivities(Set<TaskActivityType> activityTypeSet, QueryParams queryParams);
+    List<TaskActivity> getAllActivities(Set<TaskActivityType> activityTypeSet, QueryParams queryParams, boolean grouping);
```

Comment:
please rename it to 'lastExecution', grouping is to general and there is no info that it is grouping by task id